NDTensors/ext/NDTensorsCUDAExt/adapt.jl

```diff
@@ -26,3 +28,9 @@ function NDTensors.adapt_storagetype(
   BufT = buffertype(adaptor)
   return NDTensors.emptytype(NDTensors.adapt_storagetype(CuVector{ElT,BufT}, StoreT))
 end
+
+function Adapt.adapt_storage(
+  to::NDTensorsCUDAExt.NDTensorCuArrayAdaptor, x::AbstractDiagonalArray
+)
+  return DiagonalArray(adapt_storage(to, x.diag), size(x), getindex_zero_function(x))
+end
```

Comment:
This should get put in a different file so we don't start mixing code for the new and old storage systems.